R/cb_filter_apply.R

```diff
@@ -145,6 +145,26 @@
   return(column_body_all)
 }
 
+# takes any CB v2 query and recursively looks for subqueries where an
+# AND/OR operator is applied to a single condition and removes the operator. 
+# When calling this function, starting_depth should be left as default (0). Its
+# value is updated during recursion
+.extract_single_nodes <- function(x, starting_depth = 0){
+  
+  starting_depth <- starting_depth + 1
+  
+  if(!is.list(x)) return(x)
+  
+  if(
+    starting_depth > 1 &
+    !is.null(x$operator) & 
+    !identical(x$operator, "NOT") & 
+    length(x$queries) == 1
+  ) return(x$queries[[1]])
```

Comment:
```suggestion
  ) return(.extract_single_nodes(x$queries[[1]], starting_depth))
```
This will also handle the case where we somehow end up with multiple single nodes nested inside eachother.

Reply:
Nice, this is smart and would solve that situation.

(I really hope this doesn't come up too much.. many more levels of nesting will hurt my brain)